ocaml/xapi/xapi_vdi.ml

```diff
@@ -474,6 +474,7 @@ let destroy ~__context ~self =
 	let location = Db.VDI.get_location ~__context ~self in
   Sm.assert_pbd_is_plugged ~__context ~sr;
   Xapi_vdi_helpers.assert_managed ~__context ~vdi:self;
+  Xapi_vdi_helpers.assert_ha_in_progress ~__context ~vdi:self;
```

Comment:
This check should be in the VDI's allowed_operations code, otherwise the new condition won't be reflected in the VDI's allowed_operations.

The check should go in [this block](https://github.com/xapi-project/xen-api/blob/2efaa542a255550c201e8ae4c81358122fb002b9/ocaml/xapi/xapi_vdi.ml#L133-145) with the other VDI.allowed_operations checks.